mysys/my_getopt.c

```diff
@@ -1546,7 +1546,9 @@ void my_print_help(const struct my_option *options)
   for (optp= options; optp->name; optp++)
   {
     const char *typelib_help= 0;
-    if (!optp->comment)
+    if (!optp->comment &&
+        (optp->deprecation_substitute == NULL ||
+          strcmp(optp->deprecation_substitute, OPT_DEPRECATED_NO_REPLACEMENT) == 0))
```

Comment:
why you didn't use the existing convention that empty string (`""`) means no replacement?

Reply:
why does it matter here whether the option is deprecated or not? no comment means it's invisible, why would deprecation suddenly make it visible again?

Reply:
> why you didn't use the existing convention that empty string ("") means no replacement?

My goal was just to make the code more self explanatory, reading `OPT_DEPRECATED_NO_REPLACEMENT` will provide more information than `""`.

> why does it matter here whether the option is deprecated or not? no comment means it's invisible, why would deprecation suddenly make it visible again?

The idea behind it is that when a variable is just deprecated by another one, the `deprecation_substitute` will be set and the description field can be then re-used to add extra information on how to achieve the same behaviour, if needed. So if an option has been just renamed or directly replaced by another one, the `deprecation_substitute` will be set and the comment will be emptied, resulting in something like:

```
--old-variable[=name] (DEPRECATED and replaced by 'new-variable')
--new-variable[=name]  new-variable description
```

But if for example, in order to achieve the same behaviour we need to do something more specific we can add that in the comment filed, and should result in something like:

```
--old-variable[=name] (DEPRECATED and replaced by 'new-variable')
                       Use `X` value of new-variable to achieve the same behaviour.
--new-variable[=name]  new-variable description
```

Reply:
you can make the code more self explanatory without `strcmp` and `strdup`. That is, keep the existing convention where `""` means no replacement, and add something like
```c
#define DEPRECATED_NO_REPLACEMENT(X) (*(X) == 0)
```
Then you can write
```c
    if (!optp->comment &&
        (optp->deprecation_substitute == NULL ||
          DEPRECATED_NO_REPLACEMENT(optp->deprecation_substitute)))
```
which is arguably even more self explanatory than `strcmp`

Reply:
Before your changes the option with no comment was not printed by `--help`. It was an intentional feature, please, don't break it.